2020-04-introducing-mdjs-live-demos-everywhere-local-github-storybook/README.md

```diff
@@ -0,0 +1,344 @@
+
+==> INSERT SCREENSHOT <==
+
+- Add to your `package.json`:
+
+  ```json
+  "scripts": {
+    "storybook": "start-storybook --experimental-md-docs",
+  }
+  ```
+
+- Adjust your `.storybook/main.js` to load markdown files
+
+  ```js
+  const { mdjsTransformer } = require('@mdjs/core');
+
+  module.exports = {
+    stories: [
+    '../README.md',
+    '../docs/**/*.md',
+    ],
+  ```
+
+- Add to every markdown file that should be in storybook a name via
+
+  ```js
+  export default {
+    title: 'My Group/My Awesome Component',
+  };
+  ```
+
+With that, you are good to go.
+No additional changes to any files are needed a plugin will take care of it by converting your markdown files to the support mdx format of storybook.
+
+### Show it on github
+
+Github supports markdown out of the box and with this format we can go one step further.
+
+As it's not (yet - let's ask 😬) supported by github directly you will need a chrome extension called [mdjs-viewer](https://chrome.google.com/webstore/detail/mdjs-viewer/ifkkmomkjknligelmlcnakclabgohafe).
+
+You wanna see a demo without opening a different page? mdjs-viewer
+You wanna show a live example of the issue you are having? mdjs-viewer
+
+==> INSERT SCREENSHOT/GIF/VIDEO <==
+
+It sure looks like black magic.
+Install a chrome extension and suddenly github gets superpowers.
+
+All that is needed is to have Markdown files with the correct code fence blocks.
+And yeah your code needs to be able to run from within [unpkg.com](https://unpkg.com).
+
+**How does it actually work?**
+
+The extension detects on which github page you are.
+If it actually finds a markdown file or an issue with mdjs code then it adds a "show demo" button to can active it.
+Only if you click the button it will start gathering all the needed info.
+
+- find the nearest `package.json`
+- read the actual markdown file/issue content
+- replace all bare import with `unpkg.com` imports
+- replace all relative imports with `unpkg.com` and the name of the package.json + relative path for it
+- create a secured iframe
+- position the iframe absolute as an overlays
+- put the jsCode and html code inside the iframe
+- the button becomes a toggle to show/hide the iframe
+
+Some of the tasks are more complicated and require some extra work to make it secure but in the sense that's it.
+
+With that, you can put documentation with live examples on github.
+Even issues with demos showing the actual error in the code are possible.
+
+That sure sounds like a hell of a tool to improve your documentation an issue reproduction.
+Especially as the readme & issue content still remain useful even without the extension.
+
+### Supported on webcomponents.dev
+
+Fully supported by this awesome online editor.
+Give it a go and document your components in all it's glory.
+
+==> INSERT SCREENSHOT <==
+
+All the demo links are actually from [webcomponents.dev](https://webcomponents.dev/).
```

Comment:
That's maybe where you want to link the collection

https://webcomponents.dev/edit/collection/lsZ2eaviDNwy6pIBEDeL/tS7JYfymt6yeshma8Gn1?pm=1&sv=1

Reply:
thx changed 🤗